lib/azure/core/http/http_request.rb

```diff
@@ -92,19 +96,20 @@ def initialize(method, uri, options_or_body = {})
         def with_filter(filter=nil, &block)
           filter = filter || block
           if filter
-            old_impl = self._method(:call)
+            @has_retry_filter = filter.is_a? Azure::Core::Http::RetryPolicy
+            original_call = self._method(:call)
 
             # support 1.8.7 (define_singleton_method doesn't exist until 1.9.1)
```

Comment:
Minor: Do we support ruby 2.0 and below? if not we should remote this comment and try to use define_singleton_method if possible